src/Time/TimeSteppers/AdamsBashforthN.hpp

```diff
@@ -0,0 +1,188 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class AdamsBashforthN
+
+#pragma once
+
+#include <array>
+#include <boost/iterator/transform_iterator.hpp>
+#include <iterator>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Time/Time.hpp"
+#include "Time/TimeSteppers/TimeStepper.hpp"
+
+namespace TimeSteppers {
+
+/// An Nth Adams-Bashforth time stepper.
+class AdamsBashforthN : public TimeStepper::Inherit {
+ public:
+  static constexpr const size_t MAXIMUM_ORDER = 8;
+
+  // struct TargetOrder {
+  //   using type = size_t;
+  //   static constexpr OptionString_t label = {"TargetOrder"};
+  //   static constexpr OptionString_t help = {
+  //       "Target order of Adams-Bashforth method."};
+  //   static type lower_bound() { return 1; }
+  //   static type upper_bound() { return MAXIMUM_ORDER; }
+  // };
+
+  // struct SelfStart {
+  //   using type = bool;
+  //   static constexpr OptionString_t label = {"SelfStart"};
+  //   static constexpr OptionString_t help = {
+  //       "If true then the method starts at first order and increases."};
+  //   static type default_value() { return false; }
+  // };
+  // using OptionsList = tmpl::list<TargetOrder, SelfStart>;
+  // static std::string class_id() { return "AdamsBashforthN"; }
+  // static constexpr OptionString_t help = {
+  //     "An Adams-Bashforth Nth order time-stepper. The target order is the "
```

Comment:
Let's leave this as is. We can deal with formatting preferences when it's uncommented for Factory.